Automattic/jetpack
# Block fixtures: trim block fixture HTML when generating fixtures

#### Changes proposed in this Pull Request:

When a new line exists at the end of a html fixture file, the parser interprets this as an empty block. 😱 

The result is an empty block object in the parsed JSON (`*.parsed.json` files).


```
   {
        "blockName": null,
        "attrs": {},
        "innerBlocks": [],
        "innerHTML": "\n",
        "innerContent": [
            "\n"
        ]
    }
```

Trimming the output string of the HTML prevents this from occurring.

#### Jetpack product discussion
None. Unit-test framework-related.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

Check out this branch.

Find existing test fixtures, e.g., 
https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/extensions/blocks/rating-star/test/fixtures/jetpack__rating-star.html

Delete the files `test/fixtures/*.json` and `test/fixtures/*.serialized.html`

<img width="346" alt="Screen Shot 2021-03-16 at 11 26 34 pm" src="https://user-images.githubusercontent.com/6458278/111308648-0e2a8b00-86af-11eb-87b4-be4296fec05b.png">

Regenerate the fixtures, e.g., `yarn fixtures:generate extensions/blocks/rating-star/test`

Check the file contents of `jetpack__rating-star.parsed.json`

There should be no empty block object:

```
    {
        "blockName": null,
        "attrs": {},
        "innerBlocks": [],
        "innerHTML": "\n",
        "innerContent": [
            "\n"
        ]
    }
```

Check that all the tests pass: `yarn fixtures:test extensions/blocks`
